core/tracker/tracker.go

```diff
@@ -287,15 +327,49 @@ func analyseDutyFailed(duty core.Duty, allEvents map[core.Duty][]event) (bool, c
 	return true, comp, msg
 }
 
+// analyseParSigs returns a mapping partial signed data messages by peers (share index).
+func analyseParSigs(duty core.Duty, allEvents map[core.Duty][]event) parsigsByMsg {
```

Comment:
also curious as to why this function needs all the events and not just the events for the duty?
will this work?
```suggestion
func analyseParSigs(duty core.Duty, events []event) parsigsByMsg {
```

Reply:
indeed, good catch